great_expectations/profiler/parameter_builder/parameter_container.py

```diff
@@ -1,35 +1,231 @@
+    In particular, if any ParameterNode object in the tree (starting with the root-level ParameterNode object) already
+    exists, it is reused; in other words, ParameterNode objects are unique per part of fully-qualified parameter names.
+    """
+    parameter_container: ParameterContainer = ParameterContainer(parameter_nodes=None)
+    parameter_node: Optional[ParameterNode]
+    fully_qualified_parameter_name: str
+    parameter_value_details_dict: Dict[str, Any]
+    fully_qualified_parameter_name_as_list: List[str]
+    parameter_name_root: str
+    parameter_value: Any
+    parameter_details: Optional[Dict[str, Any]]
+    for (
+        fully_qualified_parameter_name,
+        parameter_value_details_dict,
+    ) in parameter_values.items():
+        validate_fully_qualified_parameter_name(
+            fully_qualified_parameter_name=fully_qualified_parameter_name
+        )
+        fully_qualified_parameter_name_as_list = fully_qualified_parameter_name[
+            1:
+        ].split(".")
+        parameter_name_root = fully_qualified_parameter_name_as_list[0]
+        parameter_value = parameter_value_details_dict["value"]
+        parameter_details = parameter_value_details_dict["details"]
+        parameter_node = parameter_container.get_parameter_node(
+            parameter_name_root=parameter_name_root
+        )
+        if parameter_node is None:
+            parameter_node = ParameterNode(
+                attributes=None, details=None, descendants=None
+            )
+            parameter_container.set_parameter_node(
+                parameter_name_root=parameter_name_root, parameter_node=parameter_node
+            )
+        _build_parameter_node_tree_for_one_parameter(
+            parameter_node=parameter_node,
+            parameter_name_as_list=fully_qualified_parameter_name_as_list,
+            parameter_value=parameter_value,
+            details=parameter_details,
+        )
+
+    return parameter_container
+
+
+def _build_parameter_node_tree_for_one_parameter(
+    parameter_node: ParameterNode,
+    parameter_name_as_list: List[str],
+    parameter_value: Any,
+    details: Optional[Any] = None,
+) -> None:
+    """
+    Recursively builds a tree of ParameterNode objects, creating new ParameterNode objects parsimoniously (i.e., only if
+    ParameterNode object, corresponding to a part of fully-qualified parameter names in a "name space" does not exist).
+    :param parameter_node: root-level ParameterNode for the sub-tree, characterized by the first parameter name in list
+    :param parameter_name_as_list: list of parts of a fully-qualified parameter name of sub-tree (or sub "name space")
+    :param parameter_value: value pertaining to the last part of the fully-qualified parameter name ("leave node")
+    :param details: metadata/details pertaining to the last part of the fully-qualified parameter name ("leave node")
```

Comment:
Sorry just a tiny nitpick
```suggestion
    :param parameter_value: value pertaining to the last part of the fully-qualified parameter name ("leaf node")
    :param details: metadata/details pertaining to the last part of the fully-qualified parameter name ("leaf node")
```